storage/innobase/dict/dict0dd.cc

```diff
@@ -6249,24 +6248,59 @@ static bool dd_update_tablespace_dd_flags(
   return (false);
 }
 
+bool dd_set_online_encryption(THD *thd, const char *space_name,
+                              volatile bool *is_space_being_removed) {
```

Comment:
Please avoid volatile.

If a barrier is needed, please use it explicitly.

Reply:
Barrier will have no use here. I do not want to order any reads/writes. The volatile pointer is_space_being_removed here reaches out to bool stop_new_ops. I want CPU to read this memory - in case it was changed. I only use it for handling case when space has been dropped or timeout on mdl rock was reached. To make sure I release the mdl in case of timeout or tablespace drop. If you do not like this volatile - I can break the wait on mdl only in case of timeout - but volatile here is a good option imho - the stop_new_ops can only go from false to true - So if the volatile bool becomes true - I am sure I can break waiting on the mdl. It has been working really well for quite some time and I would like to leave it as is.

Reply:
Ok, this is an existing change. Ideally it should keep trying until it is successful. The MDL timeouts are regular uses case. for case of DD, I dont see a reason why there would be infinte lock time..

SInce it is an existing change, ok to keep it.
Otherwise, honestly, we need a better implementation to deal with MDL timeout on DD objects.